### PR TITLE
✨ Add row wrapper component to enable external contexts like hook form

### PR DESCRIPTION
**Pull request by Simrayz**

### CODE REVIEW

File: apps/playground/src/components/PokemonTable/PokemonTable.tsx [hunk 1]
```diff
@@ -41,3 +45,9 @@ export const PokemonTable = () => {
     </div>
   )
 }
+
+const PokemonTableRow: TableRowWrapper<Pokemon> = ({ row, children }) => {
+  console.log({ rowFromPokemon: row.original.name })
```
Comment (mholta) [inline apps/playground/src/components/PokemonTable/PokemonTable.tsx:50]: Remove?

**Review by mholta — CHANGES_REQUESTED** [review]

File: apps/playground/src/components/PokemonTable/PokemonTable.tsx (hunk 1 above)
Comment (Simrayz) [reply apps/playground/src/components/PokemonTable/PokemonTable.tsx]: Fair, removed now

**Review by cbuv — CHANGES_REQUESTED** [review]
1. would be good to explain why this is needed, what use case it covers, and how this is supposed to be used. 

2. won't this break the HTML where a tr is a child of something other than tbody?

3. did you consider the react hook form issue with virtualized content? 
https://react-hook-form.com/advanced-usage#Workingwithvirtualizedlists 

Suggest we'll wait with expanding the component API until this feature is complete.

**Comment by Simrayz** [comment]
> 1. would be good to explain why this is needed, what use case it covers, and how this is supposed to be used.
> 
>     2. won't this break the HTML where a tr is a child of something other than tbody?
> 
>     3. did you consider the react hook form issue with virtualized content?
>        https://react-hook-form.com/advanced-usage#Workingwithvirtualizedlists
> 
> 
> Suggest we'll wait with expanding the component API until this feature is complete.

I considered overwriting the `TableRow` component itself, but decided on a "Wrapper" instead. However, my intention is for this to be used for something like a `<Provider />` or a Fragment. So purely for state management/wrapping the table cells in some way.

**Comment by cbuv** [comment]
> > 1. would be good to explain why this is needed, what use case it covers, and how this is supposed to be used.
> >    
> >    2. won't this break the HTML where a tr is a child of something other than tbody?
> >    3. did you consider the react hook form issue with virtualized content?
> >       https://react-hook-form.com/advanced-usage#Workingwithvirtualizedlists
> > 
> > Suggest we'll wait with expanding the component API until this feature is complete.
> 
> I considered overwriting the `TableRow` component itself, but decided on a "Wrapper" instead. However, my intention is for this to be used for something like a `<Provider />` or a Fragment. So purely for state management/wrapping the table cells in some way. @cbuv We could mark the config option as Preview/Unstable?

Yes, good idea! Marking the rowWrapper config option as unstable should let us proceed with this change.

**Review by cbuv — APPROVED** [review]

**Pull request merged**